components/react/hooks/src/useSwipe/index.js

```diff
@@ -0,0 +1,80 @@
+import {useCallback, useEffect, useState} from 'react'
+import useMount from '../useMount'
+
+const THRESHOLDS = {
+  DISTANCE: 100,
+  TIME: 100
+}
+
+export default function useSwipe(ref) {
+  const [start, setStart] = useState({x: 0, y: 0, time: 0})
+  const [capturing, setCapturing] = useState(false)
+  const [end, setEnd] = useState({x: 0, y: 0})
+  const [direction, setDirection] = useState(undefined)
+
+  const onStart = ({clientX: x, clientY: y}) => {
+    setCapturing(true)
+    setStart({x, y, time: +new Date()})
```

Comment:
I think that if we use useReducer instead useState we could improve the number of renders in this hook. Given `onStart` updates two states,  useSwipe will re-render twice for each listener update.
Did you see some problem related to that in components that uses this hook?

Reply:
It actually does not render twice: React automatically batches multiple changes in the state into once :)